Remove -fthread-jumps which causes compile failure with clang when --enable-opt
is used because clang does not understand that flag.

This patch is applied conditionally, when the compiler is not gcc.

The flag is probably also pointless for gcc since this optimization is included
in -O1 and later and this line already uses -O3.

https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html#index-fthread-jumps
--- configure.orig	2011-07-31 19:31:11.000000000 -0500
+++ configure	2024-01-10 01:43:56.000000000 -0600
@@ -14422,7 +14422,7 @@
       if test "x$enable_opt" = "xyes" -o \( "x$enable_opt" = "xmaybe" -a "x$CFLAGS" = "x" \); then
          { $as_echo "$as_me:${as_lineno-$LINENO}: result: enabled" >&5
 $as_echo "enabled" >&6; }
-         LIBJ_CFLAGS_OPT="-O3 -fomit-frame-pointer -finline-functions -funroll-loops -fthread-jumps -DNDEBUG"
+         LIBJ_CFLAGS_OPT="-O3 -fomit-frame-pointer -finline-functions -funroll-loops -DNDEBUG"
       else
          { $as_echo "$as_me:${as_lineno-$LINENO}: result: not enabled" >&5
 $as_echo "not enabled" >&6; }
